### PR TITLE
[CM-825] SPM Migration for KM Core SDK

### PR DESCRIPTION
**Pull request by Sathyan-Elangovan**
## Summary
- Added Support for SPM 
- Changed project structure for SPM
- increased pod version to 1.0.0

## Note
Locally tested by adding as SPM & pod sdk.

### CODE REVIEW

File: .swiftpm/xcode/package.xcworkspace/contents.xcworkspacedata [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Workspace
+   version = "1.0">
```
Comment (mukeshthawani) [inline .swiftpm/xcode/package.xcworkspace/contents.xcworkspacedata:3]: We can remove this file and add this file type to `.gitignore`.

File: Example/KommunicateCore-iOS-SDK.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>IDEDidComputeMac32BitWarning</key>
```
Comment (mukeshthawani) [inline Example/KommunicateCore-iOS-SDK.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:5]: See if we can remove this file.

File: Example/Podfile.lock [hunk 3]
```diff
@@ -1,5 +1,5 @@
 PODS:
-  - KommunicateCore-iOS-SDK (0.0.1)
+  - KommunicateCore-iOS-SDK (0.0.2)
```
Comment (mukeshthawani) [inline Example/Podfile.lock:2]: It should be `1.0.0`, right?

File: Package.swift [hunk 4]
```diff
@@ -3,24 +3,24 @@
 import PackageDescription
 
 let package = Package(
-    name: "KommunicateCore-iOS-SDK",
+    name: "KommunicateCore_iOS_SDK",
```
Comment (mukeshthawani) [inline Package.swift:6]: We are only changing the name in SPM, not in CP?

Comment (Sathyan-Elangovan) [reply Package.swift:6]: yes

File: Example/Podfile.lock (hunk 3 above)
Comment (Sathyan-Elangovan) [reply Example/Podfile.lock]: resolved

File: .swiftpm/xcode/package.xcworkspace/contents.xcworkspacedata (hunk 1 above)
Comment (Sathyan-Elangovan) [reply .swiftpm/xcode/package.xcworkspace/contents.xcworkspacedata]: okay will add it to .gitignore

File: Example/KommunicateCore-iOS-SDK.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (hunk 2 above)
Comment (Sathyan-Elangovan) [reply Example/KommunicateCore-iOS-SDK.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: removed

**Review by mukeshthawani — APPROVED** [review]

**Pull request merged**